arches/app/models/graph.py

```diff
@@ -1586,15 +1586,22 @@ def publish(self, notes=None):
         Adds a row to the GraphPublication table
         Assigns GraphPublication id to Graph
         """
-        publication = models.GraphPublication.objects.create(
-            graph=self,
-            serialized_graph=JSONSerializer().serialize(self, force_recalculation=True),
-            notes=notes,
-        )
-        publication.save()
+        with transaction.atomic():
+            try:
+                publication = models.GraphPublication.objects.create(
+                    graph=self,
+                    serialized_graph=JSONDeserializer().deserialize(JSONSerializer().serialize(self, force_recalculation=True)),
+                    notes=notes,
+                )
+                publication.save()
+            except Exception:
+                raise UnpublishedModelError("Could not save graph publication")
```

Comment:
Instead of raising the string, can we pass the exception here?

Reply:
good feedback, done and done.